src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/CollectionDeserializationHelpers.cs

```diff
@@ -119,10 +119,35 @@ public static void Clear(this IEnumerable collection, string propertyName, Type
             clearMethod.Invoke(collection, _emptyObjectArray);
         }
 
-        public static bool TryCreateInstance(Type collectionType, IEdmCollectionTypeReference edmCollectionType, Type elementType, out IEnumerable instance)
+        /// <summary>
+        /// Creates an instance of a collection object. 
+        /// </summary>
+        /// <param name="collectionType">The <see cref="Type"/> of the collection.</param>
+        /// <param name="edmCollectionType">The <see cref="EdmCollectionTypeReference"/> of the collection.</param>
+        /// <param name="elementType">The <see cref="Type"/> of the collection elements.</param>
+        /// <param name="instance">An instance of the collection.</param>
+        /// <param name="isDelta">true if it is a collection delta; otherwise false.</param>
+        /// <returns>true if the collection instance was successfully created; otherwise false.</returns>
+        public static bool TryCreateInstance(Type collectionType, IEdmCollectionTypeReference edmCollectionType, Type elementType, out IEnumerable instance, bool isDelta = false)
         {
             Contract.Assert(collectionType != null);
 
+            //For Delta Collection requests
+            if (isDelta)
+            {
+                if (elementType == typeof(IEdmEntityObject))
```

Comment:
Can `elementType` be an `IEdmComplexObject`?

Reply:
I believe so.